### PR TITLE
[Component with its own class name as service type] LPD-8526 Refactor Helper components from document-library-opener-google-drive-web, on-demand-admin, portal-workflow and user-associated-data-web modules

### PR DESCRIPTION
**Pull request by MrJohn1911**
Previous pull: https://github.com/liferay-core-infra/liferay-portal/pull/6680

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:workflow

**Comment by MrJohn1911** [comment]
ci:test:usm

**Comment by MrJohn1911** [comment]
ci:test:content-management

File: modules/apps/portal-workflow/portal-workflow-api/src/main/java/com/liferay/portal/workflow/portlet/tab/BaseWorkflowPortletTab.java [hunk 1]
```diff
@@ -79,6 +89,63 @@ protected Log getLog() {
 		return _logs.get(clazz);
 	}
 
+	protected String getMVCPathAttributeName(String namespace) {
```
Comment (dantewang) [inline modules/apps/portal-workflow/portal-workflow-api/src/main/java/com/liferay/portal/workflow/portlet/tab/BaseWorkflowPortletTab.java:92]: @MrJohn1911 It seems this method is only used by `getPath`, please inline if so.

File: modules/apps/portal-workflow/portal-workflow-api/src/main/java/com/liferay/portal/workflow/portlet/tab/BaseWorkflowPortletTab.java [hunk 2]
```diff
@@ -79,6 +89,63 @@ protected Log getLog() {
...
+		return StringBundler.concat(
+			namespace, StringPool.PERIOD,
+			MVCRenderConstants.MVC_PATH_REQUEST_ATTRIBUTE_NAME);
+	}
+
+	protected String getPath(
```
Comment (dantewang) [inline modules/apps/portal-workflow/portal-workflow-api/src/main/java/com/liferay/portal/workflow/portlet/tab/BaseWorkflowPortletTab.java:98]: @MrJohn1911 Some of the methods, like `getPath`, is only used by one of the sub-classes. Let's push such methods down.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by dantewang** [comment]
Plz resend, thanks.

**Pull request closed without merging**